pkg/util/bandwidth/interfaces.go

```diff
@@ -0,0 +1,23 @@
+/*
+Copyright 2014 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package bandwidth
+
+type BandwidthShaper interface {
+	InitializeInterface(iface string) error
+	LimitCIDRBandwidth(upload, download, cidr, iface string) error
```

Comment:
My first thought was to say "why not just identify the pod" - that would be a lot easier for most users.  I agree that the CIDR model is more flexible, but I suggest treating it as an extension for expert users.  (Feel free to translate to a CIDR internally, but why should the user have to do that?)

A comment or two wouldn't hurt - especially for something like the anticipated external expectations for behavior.  Can I have several of these outstanding at a time?  How many? What should I expect if the addresses overlap?  (Just FYI, I had to go look up what CIDR meant.)